drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -602,6 +602,20 @@ int stm32_i2c_configure_timing(struct device *dev, uint32_t clock)
 	uint32_t presc = 1U;
 	uint32_t timing = 0U;
 
+	/*  Look for an adequate fixed timing value */
+	for (uint32_t i = 0; i < (cfg->nr_of_timings); i++) {
+		const struct i2c_config_timing *fixed = &cfg->timings_list[i];
+		uint32_t speed = i2c_map_dt_bitrate(fixed->i2c_speed);
+
+		if ((I2C_SPEED_GET(speed) == I2C_SPEED_GET(data->dev_config))
+		   && (fixed->periph_clock == clock)) {
+			/*  Found a matching periph clock and i2c speed */
+			LL_I2C_SetTiming(i2c, fixed->timing_setting);
```

Comment:
Non blocking: 'preset' would be clearer than 'fixed'

Reply:
will do